cpp/oneapi/dal/backend/interop/csr_block_owner.hpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+namespace oneapi::dal::backend::interop {
+
+template <typename T>
+struct csr_block_owner {
+    csr_block_owner(const daal::data_management::NumericTablePtr& nt) {
+        daal::services::Status status;
+        _csr_nt = daal::services::dynamicPointerCast<daal::data_management::CSRNumericTable,
+                                                     daal::data_management::NumericTable>(nt);
+        ONEDAL_ASSERT(_csr_nt);
+        _row_count = dal::detail::integral_cast<std::int64_t>(_csr_nt->getNumberOfRows());
+        _column_count = dal::detail::integral_cast<std::int64_t>(_csr_nt->getNumberOfColumns());
+        _element_count = dal::detail::integral_cast<std::int64_t>(_csr_nt->getDataSize());
+        status = _csr_nt->getSparseBlock(0,
+                                         _csr_nt->getNumberOfRows(),
+                                         daal::data_management::readOnly,
+                                         _block);
+        status_to_exception(status);
+        _is_empty = false;
```

Comment:
The default value is not set - do we need to free the block in dtor if an error happened on line 35?

Reply:
Ok, it was removed